scriptmodules/supplementary/samba.sh

```diff
@@ -41,7 +41,8 @@ _EOF_
 }
 
 function restart_samba() {
-    service samba restart || service smbd restart
+    [[ ! -z $(systemctl list-units --full -all | grep smbd.service) ]] && service smbd restart
+    [[ ! -z $(systemctl list-units --full -all | grep samba.service) ]] && service samba restart
```

Comment:
Not sure I like the duplication. Do we really need to check - can't we just hide the error ?

``
    local service
    for service in smbd samba; do
        systemctl restart $service 2>/dev/null
    done
``

If the check is needed that could be done in the loop I guess without duplicating the command.

Reply:
Redirect output to /dev/null should work as well.